### PR TITLE
[IDB IGNORE][MDB IGNORE] Adds Jungleland as a randomly chosen alternative to Lavaland, [Bounty]

### PR DESCRIPTION
**Pull request by Djiq**
# Document the changes in your pull request
This PR adds an alternative to Lavaland that is themed after a toxic jungle on a planet brimming with sulphur and acidic pits, a small island filled with life has been found, it is insanely rich in resources, and so NT planted a base there, but the aggressive jungle fauna will not be easily defeated!

Ok this is a massive really - fucking - massive PR. First of all, Jungleland is divided into 5 Biomes

Deep Jungle - standard jungle experience, i believe it is the easiest part of the jungle as it only contains dryads and blobbies
Dying Jungle - a part of jungle that is withering away for *one* reason or another, it is a harder area, as it contains more dangerous mobs like corrupted dryads or magmawings.
Toxic Pit - A swamp filled with sulphuric water that you can easily get poisoned by, this area is quite brutal as it often spawns Medurachas and a lot of mosquitos
Rocky Beach - a transitory zone between Toxic Pits and Barren Rocks, some enemies from both
Barren Rocks - Lavaland like biomes containing watchers and goliaths.

This PR also adds unique ruins made by @Marmio64, you should be able to check them out in mapdiff, i dont really have a way to describe them, just some thematic ruins for most biomes.

Besides that this PR also aims to add a couple new vegatation and chemicals that you can gather from Jungle flora (Currently only Polybycin and Quinine are implemented, but im planning on adding more during the TM phase), some more items related to Jungleland mobs (currently the mobs unique to jungleland dont drop anything, but that is soon to be changed), and it also adds a new unique disease - Malaria Exotica, you can get it from getting bitten by Mosquitos and it is a slowly ramping up disease with periods of asymptomacity that after a couple minutes kills you by extreme fever. 

# Roadmap

- [x] Add ruins spawning
- [x] Add mining base
- [x] Add mob loot
- [x] Add megafauna spawning
- [x] Implement a megafauna unique to jungleland
- [x] Add unique vegetation along with the rest of chemicals
- [x] Add a way for admins to force either lavaland or jungleland to the following round

# Examples of Jungleland world gen
obraz
obraz
obraz

# Spriting
Once IDB issues are fixed, please look into that check to see all of the new sprites, there are literally to many to list here.

# Wiki Documentation
waiting for my work to be ported to a proper page, here is how it looks for now:
https://wiki.yogstation.net/wiki/User:Djiq

# Changelog
:cl:  
rscadd: New Mining area: Jungleland, Most content added by Djiq, Most ruins, some mobs and Ivymen added by Marmio64. Other ruins added by ToasterBiome. Various fixes and updates by courtesy of Jamie1D and SapphicOverload!
rscadd: New Mobs now appear on jungleland, such as Meduracha, Dryads, Giant mosquitoes, and Skin Twisters!
rscadd: New Mining implement: Kinetic Javelin, get it either from the roundstart kit, or buy it in a mining vendor.
rscadd: New Megafauna: Tar King - You can summon it by collecting 3 parts of a broken crystal, assembling it, and placing it in a Forgotten Altar!
rscadd: Ivymen - Jungleland's counterpart to ashwalkers, by Marmio64
rscadd: Jungleland has many new biomes, beware of sulphuric pits, as it will cause you trouble if you step into it.
imageadd: Added new icons and icon states for jungleland
mapping: Added Jungleland as a new mining destination!
/:cl:

### CODE REVIEW

**Comment by maxion12345** [comment]
IT'S TIME

**Comment by tattax** [comment]
good job

File: code/_onclick/hud/plane_master.dm [hunk 1]
```diff
@@ -59,12 +65,24 @@
...
+	//tldr dont add filters directly for the love of god, use add_filter instead.
+
+	/*filters = list()
 	if(istype(mymob) && mymob.client && mymob.client.prefs && mymob.client.prefs.ambientocclusion)
 		filters += AMBIENT_OCCLUSION
 	if(istype(mymob) && mymob.eye_blurry)
 		filters += GAUSSIAN_BLUR(clamp(mymob.eye_blurry*0.1,0.6,3))
+	*/
+
+	if(istype(mymob) && mymob.client && mymob.client.prefs && mymob.client.prefs.ambientocclusion)
```
Comment (tattax) [inline code/_onclick/hud/plane_master.dm]:
```suggestion
	if(istype(mymob) && mymob?.client?.prefs?.ambientocclusion)
```

**Comment by Djiq** [comment]
the last test is failing due to an incompatible version of rust-g being used in the CI, the one that is required for this pr to work needs to be at least 0.8.0 with --all-features flag, while the one used in the CI is downloaded from rust-g releases page and doesn't have that crucial flag

**Comment by Addust** [comment]
what about ruins

**Comment by Djiq** [comment]
> what about ruins

ruins are already included, just look through the map files, i cant be bothered to post pictures yet since im busy working on the mob loot

**Comment by Marmio64** [comment]
> what about ruins

including the tar temples and not including any ruins that just spawn megafauna, there are 13 ruins (that i know of): 11 by me and 2 by toasterbiome, and we both have pictures available in our PRs if you want to spoil the contents beforehand.
https://github.com/EdgeLordExe/Yogstation/pull/7
https://github.com/EdgeLordExe/Yogstation/pull/3
https://github.com/EdgeLordExe/Yogstation/pull/4

**Comment by Djiq** [comment]
> the last test is failing due to an incompatible version of rust-g being used in the CI, the one that is required for this pr to work needs to be at least 0.8.0 with --all-features flag, while the one used in the CI is downloaded from rust-g releases page and doesn't have that crucial flag

note to reviewers

**Comment by TheRyeGuyWhoWillNowDie** [comment]
Jungleland spawns with exactly 0 dilithium

**Comment by JamieD1** [comment]
Let me know when compile is fixed

**Comment by patpol4** [comment]
ping me when this is close to finished and you will need a thorough map review for the ruins

**Review by patpol4 — DISMISSED** [review]
I'll do some now and some later

1 jungleland_dead_bubblegum has some duplicate blood decals, is this intentional?
image
2. jungleland_dead_tartemple Replace this turf with sand, it will blend better
image

Actually during the making of this review i realized it would be easier if i just did changes to the maps to your branch directly, is that ok? There's plenty of small changes i would  like do to make it look and feel better.

**Comment by Djiq** [comment]
> Actually during the making of this review i realized it would be easier if i just did changes to the maps to your branch directly, is that ok? There's plenty of small changes i would like do to make it look and feel better.

yes it's perfectly fine by me! sorry it took me while to respond but i just simply didnt notice you commented here

**Comment by Moltijoe** [comment]
rip

**Comment by Djiq** [comment]
> rip

project on standby till i finish my exams in may

**Comment by N3D6** [comment]
Alright, cool concept but it is just not enjoyable whatsoever to play right now. I like the idea of alternating mines but this needs a lot of work before it should even be considered for a merge

**Comment by MajManatee** [comment]
I have a dream. That one day, man will have jungle land added as a random chance map.

**Comment by Djiq** [comment]
Adding [IDB IGNORE] till the bug is resolved

**Comment by JohnFulpWillard** [comment]
I think you got a merge conflict

image

**Comment by Djiq** [comment]
i was asleep ;_;

**Comment by Cartlord** [comment]
I played this, and honestly, it just feels... weird. The resources are available in such massive quantities that it feels way too easy to mine everything the station needs in a very short window of time, and the bug where mining satchels don't automaticly pick anything up unless you drop them and re-equip them is really unusual.

**Review by ToasterBiome — APPROVED** [review]

**Comment by ToasterBiome** [comment]
well technically the entire bounty is fulfilled and you plan on supporting nit for a while and fixing anything that pops up.. time to merge

**Comment by MajManatee** [comment]
a historic day for yogskind

**Pull request merged**